src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/StringGuid.cs

```diff
@@ -7,7 +7,7 @@
 namespace Microsoft.MixedReality.SpectatorView
 {
     [Serializable]
-    internal class StringGuid
+    internal class StringGuid : IComparable

```

Comment:
>StringGuid [](start = 19, length = 10)

Out of curiosity, what's wrong with the ordinary GUID? It's already comparable, and should be way faster to work with.

Reply:
I'm not sure historically why the GUID wasn't used. is a GUID serializable?

Reply:
The docs say that it [has](https://docs.microsoft.com/en-us/dotnet/api/system.guid?view=netframework-4.8) SerializableAttribute, so it should be.

I'm genuinely curious why the string representation was chosen instead. Maybe there was some issue with the Unity integration?